sdk/java/pulumi/src/test/java/com/pulumi/core/UrnTest.java

```diff
@@ -20,7 +20,7 @@ public static void mockSetup() {
         mock = DeploymentTests.DeploymentMockBuilder.builder()
             .setMocks(new MocksTest.MyMocks())
             .setOptions(new TestOptions(false))
-            .setMockGlobalInstance();
+            .build();
```

Comment:
Does it not actually set the global instance still? The previous name was more suggestive.

Reply:
yes, yes it does, a builder not ending with build feels weird, as a target API that I'm working on I was thinking about:
```
PulumiTest.withOptions(new TestOptions(false))
  .monitorMocks(new MocksTest.MyMocks())
  .useMockRunner()
  .build()
```
The fact that we set a global singleton is an implementation detail that is also true for the `Pulumi.run` itself, and not specific to the test API, so we should document it but making is as part of API does not feel like thinking ahead, esp. since we may change this in the future.

Reply:
OK then.